### ISSUE TITLE
Hello world is too complicated for 1st timer

### ISSUE THREAD

**Issue by isidore** [exercism/swift#378]
tl:dr;
The hello world should start with 
``` swift
func hello(_ name: String? = nil) -> String{
return "fix this"
}
```

## Background
I am a very advanced programmer. I started the swift course today with my girlfriend as she wanted to learn swift (her team uses it). It was remarkably confusing. In the end, we googled a solution(which didn't work) submitted it anyways. Then looked at the other solutions to get the answer. This process **took about 3 hours** all said and done

### Complexities in the Method signature

``` swift
func hello(_ name: String? = nil) -> String{
return ""
}
```
To get this to compile. you have to write this method. Let's look at the pieces

### Basic signature
``` swift
func hello(name: String) -> String
```
I felt that this is reasonable. The documentation has pieces that show how to make a method. but it does not work. The compiler errors are confusing so I thought it might need a class

``` swift
class HelloWorld{
func hello(name: String) -> String
}
```
this makes things worse.

### The underscore
```swift
_
```
This means that you don't have to call the method with the parameter label.
Which you don't. So removing it breaks the build. While I've seen many languages that have optional parameter names, this is the first I've seen where the default is non-optional. optional parameter names is a decently complex language feature to have to understand for hello world.

### The Question Mark ?
```swift
 String?
```
Nullable strings (or Nilable?) because you are passing in nil (you actually aren't but effectively are by not declaring the variable)  means you have to understand nil and ? both of which are a bit complex and ? is hard to google, although seeing it from the you just have to know to do this first, then google. it's not something we would have ever googled.

I tried to do this with method overloading (writing two method) but it didn't work, possibly because they were in a class at the time, I don't remember.

### The default value 
``` swift
= nil
```
This is an other fairly advanced concept. That if you don't pass in a value one is taken care of for you. 

## Feedback while failing
All of this feedback is at the compiler level, because until it compiles you get nothing. Compiler messages are confusing. And there is no help for the brand new swift programmer. 

This could also be solved by changing the test to require a less robust signature in the function. just removing the null case and adding the named parameters would lead to

``` swift
func hello(name: String) -> String{
return "fix this"
}
```
Which I (as an advanced program) fell I could have reasonable figured out.
But I still think you should just include the method signature you want. I don't think that is what you are trying to focus on at the moment for Hello world.

**Comment by isidore** [comment]
ps, if this bug report seems a bit too long for hello world, that is kind of my point.

**Comment by masters3d** [comment]
@isidore I agree. Can you submit a PR with this change?

**Comment by isidore** [comment]
I can, but would prefer not to for just one line, do you mind doing it?

**Comment by masters3d** [comment]
Related. https://github.com/exercism/exercism/issues/4707#issuecomment-491575304

**Comment by darya-yilmaz** [comment]
What to write in a code to pass test?

**Comment by kotp** [comment]
When the file first opens in the editor, and the tests have ran, what do the tests say?  It should almost give the answer there.  The code given, the string itself, is what is causing it to not pass.  We may write the string that the test run is reporting it should be, and it should pass.

I hope that helps, @darya-yilmaz .

**Comment by kotp** [comment]
Since the source given, the following:

```swift
func hello() -> String {
  return "Goodbye, Mars!"
}
```

Is the code delivered now, I think this can be closed.
